### PR TITLE
feat(mc): #3262 Add placeholders for TopSites

### PR DESCRIPTION
**Pull request by k88hudson**
This adds placeholders for TopSites. Part of the prerendering work.

This is alternative approach that requires fewer changes, which one do you think is better?

https://github.com/mozilla/activity-stream/compare/master...k88hudson:gh3262b?expand=1

### CODE REVIEW

File: system-addon/content-src/components/TopSites/TopSite.jsx [hunk 1]
```diff
@@ -5,6 +5,41 @@ const LinkMenu = require("content-src/components/LinkMenu/LinkMenu");
 
 const {TOP_SITES_SOURCE, TOP_SITES_CONTEXT_MENU_OPTIONS} = require("./TopSitesConstants");
 
+const TopSiteLink = props => {
+  const {link} = props;
+  const topSiteOuterClassName = `top-site-outer${props.className ? ` ${props.className}` : ""}`;
```
Comment (rlr) [inline system-addon/content-src/components/TopSites/TopSite.jsx]: crazy that you can have template literal inside a template literal.

File: system-addon/content-src/components/TopSites/TopSites.jsx [hunk 2]
```diff
@@ -4,22 +4,27 @@ const {FormattedMessage} = require("react-intl");
...
-        link={link}
-        index={index}
-        intl={props.intl} />)}
-    </ul>
-    <TopSitesEdit {...props} />
-  </section>
-</TopSitesPerfTimer>);
+const TopSites = props => {
+  const realTopSites = props.TopSites.rows.slice(0, props.TopSitesCount);
+  const placeholderCount = props.TopSitesCount - realTopSites.length;
```
Comment (rlr) [inline system-addon/content-src/components/TopSites/TopSites.jsx]: Do we no longer allow "holes" in the top sites? Say the user pinned something in the very last slot (#12) but only has 6 other top sites. What does the UI show for that? Does it collapse it? I guess I can test it out.

**Comment by as-pine-proxy** [comment]
[Treeherder: a637b4d8a29f85b2e0dadfccae2843fed096640b](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=a637b4d8a29f85b2e0dadfccae2843fed096640b)

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 3]
```diff
@@ -135,7 +144,7 @@
 
     .title {
       font: message-box;
-      height: 20px;
+      height: $top-sites-title-height;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Was this change intentional? The spec shows about 67px from the bottom of the top site square to the top of the Pocket logo, but with this change, there's now about 72px. (I.e., the height was 10px shorter than the line height resulting in 5px closer square.)

before/expected: image vs image :now
